src/cmd/encoder/encoding/encoding.go

```diff
@@ -69,33 +87,62 @@ func fetchVideoSource(s3Client clients.IS3Client, videoData *contracts.Video) er
 }
 
 func encode(data *contracts.Video) error {
-	withSound, err := ffmpeg.CheckContainsSound(data.GetSource())
+	sourcefile := filepath.Base(data.GetSource())
+
+	withSound, err := ffmpeg.CheckContainsSound(sourcefile)
 	if err != nil {
 		return err
 	}
 	if !withSound {
-		if err := ffmpeg.AddEmptyAudioTrack(data.GetSource()); err != nil {
+		if err := ffmpeg.AddEmptyAudioTrack(sourcefile); err != nil {
 			return err
 		}
 	}
 
-	res, err := ffmpeg.ExtractResolution(data.GetSource())
+	res, err := ffmpeg.ExtractResolution(sourcefile)
 	if err != nil {
 		return err
 	}
-	if err = ffmpeg.ConvertToHLS(data.GetSource(), res); err != nil {
+	if err = ffmpeg.ConvertToHLS(sourcefile, res); err != nil {
+		return err
+	}
+	return nil
+}
+
+func fetchCoverSource(s3Client clients.IS3Client, videoData *contracts.Video) (isFileFetch bool, err error) {
+	if filepath.Ext(videoData.GetCoverPath()) != ".png" {
```

Comment:
Then we compress only .png files.
So it is possible to upload 15Mo jpg files.
In the same way, we never check the resulting file size, may be we want a limit of 1-2Mo per cover. Since we list 10 videos on the home page, it's already 10-20Mo to download for the client side.
It's possible to select a quality percentage with imageMagick lib, maybe we can do so with ffmpeg.
If you want to work on it in another PR, we can create an issue for it.

Reply:
Yes I was aware of this problem but as you said, I thought about doing it in another PR because it would make that one too big. I thought it could be 2 solutions to this issue:
- Check / Limit the size of the file
- Downsize the ratio scale (ex: 1920x1080) of any file to a fixed ratio (let's say 320px wide). I don't know how to technically do it right now, but the goal would be to shrink the image size by playing with its ratio scale.